lucille-plugins/lucille-ocr/src/main/java/com/kmwllc/lucille/ocr/stage/ApplyOCR.java

```diff
@@ -0,0 +1,307 @@
+ * </p>
+ * <p>
+ * <b>pages_field</b> (String, Optional) : Field on document containing a JSON string used for dynamically applying templates to documents.
+ * This allows a user to specify on the document itself which types of forms the document is holding and on which pages. The JSON should be a map 
+ * from page numbers to template names. If a page appears in both the static and dynamic mapping, the dynamic one takes precedence.
+ * </p>
+ */
+public class ApplyOCR extends Stage {
+
+  public static final String TEMP_DIR = "luclle-ocr-temp";
+  private static final Logger log = LoggerFactory.getLogger(ApplyOCR.class);
+
+  private final String lang;
+  private transient TessBaseAPI api = null;
+  private final String pathField;
+  private final String pagesField;
+  private final Map<Integer, String> pages;
+  private final Map<String, FormTemplate> extractionTemplates;
+  private final String extractAllDest;
+
+
+  public ApplyOCR(Config config) throws StageException {
+    super(config, new StageSpec().withOptionalProperties("pages_field", "extraction_templates", "extract_all_dest")
+        .withRequiredProperties("lang", "path_field").withOptionalParents("pages"));
+
+    lang = config.getString("lang");
+    pathField = ConfigUtils.getOrDefault(config, "path_field", null);
+    pagesField = ConfigUtils.getOrDefault(config, "pages_field", null);
+    extractAllDest = ConfigUtils.getOrDefault(config, "extract_all_dest", null);
+
+    if (config.hasPath("extraction_templates")) {
+      extractionTemplates = new LinkedHashMap<>();
+      List<FormTemplate> temp = config.getConfigList("extraction_templates").stream()
+          .map((c) -> ConfigBeanFactory.create(c, FormTemplate.class)).collect(Collectors.toList());
+      for (FormTemplate template : temp) {
+        extractionTemplates.put(template.getName(), template);
+      }
+    } else {
+      extractionTemplates = null;
+    }
+
+    if (config.hasPath("pages")) {
+      Map<String, Object> temp = config.getConfig("pages").root().unwrapped();
+      pages = new LinkedHashMap<>();
+      for (Map.Entry<String, Object> entry : temp.entrySet()) {
+        pages.put(Integer.parseInt(entry.getKey()), (String) entry.getValue());
+      }
+    } else {
+      pages = null;
+    }
+
+    if ((pages != null || pagesField != null) && extractionTemplates == null) {
+      throw new StageException("extraction_templates must be specified when pages or pages_field is specified");
+    }
+  }
+
+  @Override
+  public void start() throws StageException {
+    api = new TessBaseAPI();
+    String tessData = "TesseractOcr";
+    // load the models
+    if (api.Init(tessData, lang) != 0) {
+      throw new StageException(String.format("Unable to load tesseract model: %s", lang));
+    }
+  }
+
+  private String ocr(BufferedImage image) {
+    // The tesseract api has some issues reading the image directly as a byte
```

Comment:
did we confirm that the latest tesseract still has this problem?

Reply:
making a ticket to address this in the future